pincer/objects/guild/channel.py

```diff
@@ -645,21 +616,364 @@ async def create_invite(
+            The user to add to the thread.
+        """
+        await self._http.put(f"channels/{self.id}/thread-members/{user.id}")
+
+    async def leave_thread(self):
+        """|coro|
+        Removes the current user from a thread.
+        Also requires the thread to not be archived.
+        """
+        await self._http.delete(f"channels/{self.id}/thread-members/@me")
+
+    async def remove_thread_member(self, user: User):
+        """|coro|
+        Removes another member from a thread. Requires the ``MANAGE_THREADS``
+        permission, or the creator of the thread if it is a
+        ``GUILD_PRIVATE_THREAD``. Also requires the thread to not be archived.
+
+        Parameters
+        ----------
+        user: :class:`~pincer.objects.user.User`
+            The user to remove from the thread.
+        """
+        await self._http.delete(f"channels/{self.id}/thread-members/{user.id}")
+
+    async def get_thread_member(self, user: User) -> ThreadMember:
+        """|coro|
+        Returns a thread member object for the specified user if they are a
+        member of the thread.
+
+        Parameters
+        ----------
+        user: :class:`~pincer.objects.user.User`
+            The user to get the thread member for.
+
+        Returns
+        -------
+        :class:`~pincer.objects.channel.ThreadMember`
+            The thread member object.
+        """
+        return ThreadMember.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.get(
+                    f"channels/{self.id}/thread-members/{user.id}"
+                ),
+            )
+        )
+
+    async def list_thread_members(self) -> AsyncIterator[ThreadMember]:
+        """|coro|
+        Fetches all the thread members for the thread. Returns an iterator of
+        ThreadMember objects.
+
+        Returns
+        -------
+        AsyncIterator[:class:`~pincer.objects.channel.ThreadMember`]
+            An iterator of thread members.
+        """
+        data = await self._http.get(f"channels/{self.id}/thread-members")
+        for member in data:
+            yield ThreadMember.from_dict(
+                construct_client_dict(self._client, member)
+            )
+
+    async def list_active_threads(self) -> Dict[str, Any]:
+        """|coro|
+        Returns all active threads in the channel, including public and
+        private threads. Threads are ordered by their id, in descending order.
+
+        Returns
+        -------
+        Dict[:class:`str`, :class:`Any`]
+            Dictionary containing the response body:
+                - threads: Generator of the active threads.
+                - members: Generator for each thread member object for each
+                returned thread the current user has joined.
+                - has_more: Whether there are potentially additional threads
+                that could be returned on a subsequent call.
```

Comment:
The return type just needs to be fixed to be a `ThreadsResponse` object :rocket: